### ISSUE TITLE
Record Stream

### ISSUE THREAD

**Issue by mathieuHa** [phoboslab/jsmpeg#142]
Hi,

I've manage to get the stream working well, (without any control) but It does what I want.
I'm Running it from a Raspberry Pi using the onboard raspicam v2.1 and I would like to be able to record the stream or take pictures while it's streaming.

If possible with the ability to customize the size and quality.
I have some errors when I try to use the camera which is already in use.

Thank you by advance,

**Comment by szhansen** [comment]
hi, how to record the video  ?

**Comment by olegIvani4** [comment]
use `ffmpeg` for stream recording.

**Comment by szhansen** [comment]
hi,oleg,could you tell me the detail ?

**Comment by szhansen** [comment]
I mean how could you use ffmpeg to decode the video and at the same time to use it to record ?

**Comment by mathieuHa** [comment]
Yes @szhansen  that was exactly what I wanted to do, for now I haven't found a solution

**Comment by olegIvani4** [comment]
you have to run two `ffmpeg` processes. One for mpeg1 transcoding and send for saving stream to local files. e.g.
```
ffmpeg -i rtsp://10.2.2.19/live/ch01_0 -c copy -map 0 -f segment -segment_time 300 -segment_format mp4 "capture-%03d.mp4"
```

**Comment by mathieuHa** [comment]
Thank @olegIvani4 you I will try that

**Comment by phoboslab** [comment]
You can also set `RECORD_STREAM = true` in the [websocket-relay.js, line 22](https://github.com/phoboslab/jsmpeg/blob/master/websocket-relay.js#L22). This will record the incoming mpegts stream to the `recordings/` directory (you have to create this directory first).

**Comment by alanfoxye** [comment]
Wonderful work! I tried this new version with audio on Raspberry Pi. But I wonder can I still use jsmpeg.startRecording() and jsmpeg.stopRecording() to record stream from web client as the original version? @phoboslab  Thanks a lot!

**Comment by szhansen** [comment]
@alanfoxye how about the jsmpeg.startRecording() and jsmpeg.stopRecording() ,
it work ?
